pkgs/desktops/gnome/extensions/gjs-osk/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  jq,
+}:
+
+let
+  uuid = "gjsosk@vishram1123.com";
+in
+stdenv.mkDerivation rec {
+  pname = "gjs-osk";
+  version = "unstable-2024-09-24";
+
+  src = fetchFromGitHub {
+    owner = "Vishram1123";
+    repo = pname;
+    rev = "2506a863b277339805519bca06d1b101cf92eb32";
+    sha256 = "sha256-AogDo0A50SLY2YiMdtXTG/j1m3Zgu9CMTxJjNyWu4bE=";
+  };
+
+  nativeBuildInputs = [ jq ];
+
+  passthru = {
+    extensionUuid = uuid;
+    extensionPortalSlug = pname;
+  };
+
+  patchPhase = ''
+    runHook prePatch
+    pushd ${uuid}
+
+    # gjs-osk is compatible with gnome 47
+    METADATA="$(cat metadata.json)"
+    echo "$METADATA" | jq '."shell-version" += ["47"]' > metadata.json
+
+    # gjs-osk will unpack keycodes.tar.xz to store at runtime if keycodes does not exist.
+    mkdir keycodes
+    tar xf keycodes.tar.xz -C keycodes
+
+    popd
+    runHook postPatch
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    DIR=$out/share/gnome-shell/extensions
+    mkdir -p $DIR
+    cp -r ${uuid} $DIR
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A (marginally) better on screen keyboard for GNOME 42+";
+    license = licenses.gpl3;
```

Comment:
`gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.